spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/configclient/ZookeeperConfigServerBootstrapper.java

```diff
@@ -96,30 +96,31 @@ public void initialize(BootstrapRegistry registry) {
 			return new DefaultServiceDiscoveryCustomizer(curator, properties, serializer);
 		});
 		registry.registerIfAbsent(ServiceDiscovery.class, context -> {
-			if (!isEnabled(context.get(Binder.class))) {
+			if (!isEnabled(context)) {
 				return null;
 			}
 			ServiceDiscoveryCustomizer customizer = context.get(ServiceDiscoveryCustomizer.class);
 			return customizer.customize(ServiceDiscoveryBuilder.builder(ZookeeperInstance.class));
 		});
 		registry.registerIfAbsent(ZookeeperDiscoveryClient.class, context -> {
-			Binder binder = context.get(Binder.class);
-			if (!isEnabled(binder)) {
+			PropertyResolver propertyResolver = getPropertyResolver(context);
+			if (!isEnabled(context)) {
 				return null;
 			}
```

Comment:
```suggestion
			if (!isEnabled(context)) {
				return null;
			}
			PropertyResolver propertyResolver = getPropertyResolver(context);
```
(Same as the other `getPropertyResolver` call above)